### ISSUE TITLE
Callback implementation to monitor confirmation emails sent by VA Notify

### ISSUE THREAD

**Issue by baltes** [department-of-veterans-affairs/va.gov-team#98887]
As a member of the IVC forms team, I want a way to monitor the confirmation emails that are sent for a given form so that I have visibility into their status.

### Acceptance criteria
- [ ] I can see aggregate numbers for a given span of time showing 
   - [ ] Total count of emails sent
   - [ ] Successful deliveries
   - [ ] Failed deliveries

### Implementation information
https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/va_notify/README.md

**Comment by michaelclement** [comment]
Work is ongoing in [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/20334).

**Comment by michaelclement** [comment]
Waiting on engineering review of the first attempt - I can't properly test this until it's merged into staging.

**Comment by michaelclement** [comment]
Seeing some flakiness on the tests in CI, trying to determine if they're a result of this PR or something else. The error messages seem unrelated.

**Comment by michaelclement** [comment]
Verified callback works in staging. Planning on putting together new Data Dog charts this afternoon.

**Comment by michaelclement** [comment]
The callback doesn't seem to be working in production. I asked a question about rate limiting in the VA notify channel and haven't heard back. I'll plan to ping them again tomorrow to see if there's anything I need to configure.

**Comment by michaelclement** [comment]
Still waiting to hear back from VA Notify folk. This is the [dashboard](https://vagov.ddog-gov.com/dashboard/6eg-fke-mxf/ivc-champva-testing-do-not-use?fromUser=true&refresh_mode=sliding&from_ts=1737470233707&to_ts=1737643033707&live=true) where I've been testing the new email delivery graphs.

**Comment by michaelclement** [comment]
@baltes This functionality is working. You can verify by checking out this test [dashboard](https://vagov.ddog-gov.com/dashboard/6eg-fke-mxf/ivc-champva-testing-do-not-use?fromUser=true&refresh_mode=sliding&tpl_var_env%5B0%5D=eks-prod&from_ts=1739301806416&to_ts=1739316206416&live=true).

Looks like the functionality came online around 4pm - you can see successful email callbacks triggering for 10-10d submissions (no FMP1 yet) :

<img width="1352" alt="Image" src="https://github.com/user-attachments/assets/7e94a7cc-9734-4998-91c5-7b68ee4fb2d7" />
